{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from glob import glob\n",
    "from tqdm import tqdm\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import natsort as ns\n",
    "import numpy as np\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/pd378/miniconda3/envs/jupyter_env/lib/python3.6/site-packages/ipykernel_launcher.py:22: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n"
     ]
    }
   ],
   "source": [
    "#Define my directories and location of files\n",
    "WRK = '/workdir/users/pd378/oilPCR/bulk_process_all_OIL_for_paper/'\n",
    "\n",
    "F_LIST = WRK+'file_lists/filelist.txt' #a list of the files used for iterating and namings6\n",
    "\n",
    "#This table connects the second round of uniq reads witht he first round sample specific uniqs\n",
    "UNIQ_TAB=WRK+\"s6_cluster_otu/filtered_uniqs/combined_uniq_tab.txt\"\n",
    "#the parsed tabels generated from the usearch otu clustering \n",
    "P_TABLE=WRK+'s6_cluster_otu/filtered_uniqs/uniq_parsed_otu_tab.txt'\n",
    "#The table of taxonomy generated from using mother to assign taxonomy\n",
    "TAX_TAB=WRK+'s6_cluster_otu/filtered_uniqs/uniq_clustered_otus.V4_oil.wang.taxonomy' \n",
    "#the blast table generated from blasting the targets\n",
    "BLAST_TABS=WRK+'s7_blast_targets/blast_reults/'            \n",
    "#where this all will go\n",
    "OUT_TAB=WRK+\"s8_compile_data/\"\n",
    "\n",
    "\n",
    "#inport all the data\n",
    "\n",
    "uniq_df = pd.read_csv(UNIQ_TAB, sep='\\t', names=['output_uniq','cluster_num','member_num','output_size','input_first'])\n",
    "parse_df = pd.read_csv(P_TABLE, sep='\\t', names=['type','match'])\n",
    "tax_df = pd.read_csv(TAX_TAB, sep='[\\t|;]', names=['Kingdom','Phylum','Class','Order','Family','Genus','Species'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#split up the uniq_df for easier reading\n",
    "file_list=[]\n",
    "size_list=[]\n",
    "uniq_list=[]\n",
    "for item in uniq_df.index:\n",
    "    path=item.split(\"/\")\n",
    "    file_list.append(path[8].split(\";\")[0].strip('_ribo.fa'))\n",
    "    size_list.append(path[8].rsplit(\";\")[2].strip('size='))\n",
    "    uniq_list.append(path[8].split(\";\")[1])\n",
    "uniq_df['file'] = file_list\n",
    "uniq_df['in_size']=size_list\n",
    "uniq_df['in_uniq']=uniq_list\n",
    "uniq_df.sort_index\n",
    "\n",
    "\n",
    "#update the parsed file to make more sense\n",
    "index_list=[]\n",
    "sample_list=[]\n",
    "size_list=[]\n",
    "otu_liste=[]\n",
    "for item in parse_df.index:\n",
    "    name=item.split(';')[0]\n",
    "    index_list.append(name)\n",
    "    size_list.append(item.split(';')[1].strip(\"size=\"))\n",
    "    sample_list.append(name.split(\"_Uniq\")[0])\n",
    "   \n",
    "parse_df.index = index_list\n",
    "parse_df[\"size\"]=size_list\n",
    "parse_df[\"sample\"]=sample_list\n",
    "\n",
    "\n",
    "\n",
    "#parse through the ky table and pull out the otu name\n",
    "#then use the otu to fetch the taxonomy from the tax_df\n",
    "otu_list=[]\n",
    "for row in parse_df.iterrows():\n",
    "    otu = \"\"\n",
    "    if re.findall(\"otu\\d+\", row[1][0]):\n",
    "        otu = row[1][0]\n",
    "    elif \"chimera\" in (row[1][0]) or \"Chimera\" in row[1][1]:\n",
    "        otu = \"chimera\"\n",
    "    else:\n",
    "        otu = re.findall(\"otu\\d+\",row[1][1])[0]\n",
    "    otu_list.append(otu)\n",
    "    \n",
    "parse_df[\"otu\"]=otu_list\n",
    "\n",
    "# merge the parse and taxa table together\n",
    "key_df=pd.merge(parse_df, tax_df, left_on=\"otu\", how='outer', right_on=tax_df.index)\n",
    "\n",
    "# continue merging the parse_df and tax_df with the uniq_df\n",
    "cross_df=pd.merge(key_df,uniq_df,left_on='sample',right_on='output_uniq', how='outer', indicator=True)\n",
    "\n",
    "#clean up the resulting df \n",
    "clean_df=cross_df[['file', 'in_size','in_uniq','otu', 'Kingdom', 'Phylum', 'Class','Order', 'Family', 'Genus']]\n",
    "\n",
    "#strip off the weight from the genus so we can consolidate results better"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unable to open blast table 053-Single-B335-1_CTX-M-A\n",
      "Unable to open blast table 054-Single-B335-2_CTX-M-A\n",
      "Unable to open blast table 055-Single-B335-3_CTX-M-A\n",
      "Unable to open blast table 056-Single-NEG_CTX-M-A\n",
      "Unable to open blast table 057-Single-B335-1_TEM-A\n",
      "Unable to open blast table 058-Single-B335-2_TEM-A\n",
      "Unable to open blast table 059-Single-B335-3_TEM-A\n",
      "Unable to open blast table 060-Single-NEG_TEM-A\n",
      "Unable to open blast table 061-Single-B335-1_CTX-M-B\n",
      "Unable to open blast table 062-Single-B335-2_CTX-M-B\n",
      "Unable to open blast table 063-Single-B335-3_CTX-M-B\n",
      "Unable to open blast table 064-Single-NEG_CTX-M-B\n",
      "Unable to open blast table 065-Single-B335-1_TEM-B\n",
      "Unable to open blast table 066-Single-B335-2_TEM-B\n",
      "Unable to open blast table 067-Single-B335-3_TEM-B\n",
      "Unable to open blast table 068-Single-NEG_TEM-B\n",
      "Unable to open blast table 069-Single-B335-1_CTX-M-C\n",
      "Unable to open blast table 070-Single-B335-2_CTX-M-C\n",
      "Unable to open blast table 071-Single-B335-3_CTX-M-C\n",
      "Unable to open blast table 072-Single-NEG_CTX-M-C\n",
      "Unable to open blast table 073-Single-B335-1_TEM-C\n",
      "Unable to open blast table 074-Single-B335-2_TEM-C\n",
      "Unable to open blast table 075-Single-B335-3_TEM-C\n",
      "Unable to open blast table 076-Single-NEG_TEM-C\n",
      "Unable to open blast table 077-Single-B335-1_SHV-A\n",
      "Unable to open blast table 078-Single-B335-2_SHV-A\n",
      "Unable to open blast table 079-Single-B335-3_SHV-A\n",
      "Unable to open blast table 080-Single-NEG_SHV-A\n",
      "Unable to open blast table 081-Single-B335-1_Tn-ISE-A\n",
      "Unable to open blast table 082-Single-B335-2_Tn-ISE-A\n",
      "Unable to open blast table 083-Single-B335-3_Tn-ISE-A\n",
      "Unable to open blast table 084-Single-NEG_Tn-ISE-A\n",
      "Unable to open blast table 085-Single-B335-1_SHV-B\n",
      "Unable to open blast table 086-Single-B335-2_SHV-B\n",
      "Unable to open blast table 087-Single-B335-3_SHV-B\n",
      "Unable to open blast table 088-Single-NEG_SHV-B\n",
      "Unable to open blast table 089-Single-B335-1_Tn-ISE-B\n",
      "Unable to open blast table 090-Single-B335-2_Tn-ISE-B\n",
      "Unable to open blast table 091-Single-B335-3_Tn-ISE-B\n",
      "Unable to open blast table 092-Single-NEG_Tn-ISE-B\n",
      "Unable to open blast table 093-Single-B335-1_SHV-C\n",
      "Unable to open blast table 094-Single-B335-2_SHV-C\n",
      "Unable to open blast table 095-Single-B335-3_SHV-C\n",
      "Unable to open blast table 096-Single-NEG_SHV-C\n",
      "Unable to open blast table 097-TransP-B335-1_Tn-ISE-C\n",
      "Unable to open blast table 098-TransP-B335-2_Tn-ISE-C\n",
      "Unable to open blast table 099-TransP-B335-3_Tn-ISE-C\n",
      "Unable to open blast table 100-TransP-NEG_Tn-ISE-C\n",
      "Unable to open blast table 101-TransP-B335-1_Tn-Reso-A\n",
      "Unable to open blast table 102-TransP-B335-2_Tn-Reso-A\n",
      "Unable to open blast table 103-TransP-B335-3_Tn-Reso-A\n",
      "Unable to open blast table 104-TransP-NEG_Tn-Reso-A\n",
      "Unable to open blast table 105-TransP-B335-1_Tn-Tran-A\n",
      "Unable to open blast table 106-TransP-B335-2_Tn-Tran-A\n",
      "Unable to open blast table 107-TransP-B335-3_Tn-Tran-A\n",
      "Unable to open blast table 108-TransP-NEG_Tn-Tran-A\n",
      "Unable to open blast table 109-TransP-B335-1_Tn-Reso-B\n",
      "Unable to open blast table 110-TransP-B335-2_Tn-Reso-B\n",
      "Unable to open blast table 111-TransP-B335-3_Tn-Reso-B\n",
      "Unable to open blast table 112-TransP-NEG_Tn-Reso-B\n",
      "Unable to open blast table 113-TransP-B335-1_Tn-Tran-B\n",
      "Unable to open blast table 114-TransP-B335-2_Tn-Tran-B\n",
      "Unable to open blast table 115-TransP-B335-3_Tn-Tran-B\n",
      "Unable to open blast table 116-TransP-NEG_Tn-Tran-B\n",
      "Unable to open blast table 117-TransP-B335-1_Tn-Reso-C\n",
      "Unable to open blast table 118-TransP-B335-2_Tn-Reso-C\n",
      "Unable to open blast table 119-TransP-B335-3_Tn-Reso-C\n",
      "Unable to open blast table 120-TransP-NEG_Tn-Reso-C\n",
      "Unable to open blast table 121-TransP-B335-1_Tn-Tran-C\n",
      "Unable to open blast table 122-TransP-B335-2_Tn-Tran-C\n",
      "Unable to open blast table 123-TransP-B335-3_Tn-Tran-C\n",
      "Unable to open blast table 124-TransP-NEG_Tn-Tran-C\n",
      "Unable to open blast table 125-Romb-B335-1_Romb-A\n",
      "Unable to open blast table 126-Romb-B335-2_Romb-A\n",
      "Unable to open blast table 127-Romb-B335-3_Romb-A\n",
      "Unable to open blast table 128-Romb-NEG_Romb-A\n",
      "Unable to open blast table 129-Romb-B335-1_Romb-B\n",
      "Unable to open blast table 130-Romb-B335-2_Romb-B\n",
      "Unable to open blast table 131-Romb-B335-3_Romb-B\n",
      "Unable to open blast table 132-Romb-NEG_Romb-B\n",
      "Unable to open blast table 133-Romb-B335-1_Romb-C\n",
      "Unable to open blast table 134-Romb-B335-2_Romb-C\n",
      "Unable to open blast table 135-Romb-B335-3_Romb-C\n",
      "Unable to open blast table 136-Romb-NEG_Romb-C\n",
      "Unable to open blast table 137-Single-CTX-NEG-A\n",
      "Unable to open blast table 138-Single-TEM-NEG-A\n",
      "Unable to open blast table 139-Single-SHV-NEG-A\n",
      "Unable to open blast table 140-TransP-ISE-NEG-A\n",
      "Unable to open blast table 141-TransP-Reso-NEG-A\n",
      "Unable to open blast table 142-TransP-Tran-NEG-A\n",
      "Unable to open blast table 143-Romb-Romb-NEG-A\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/pd378/.local/lib/python3.6/site-packages/pandas/core/frame.py:3781: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  return super(DataFrame, self).rename(**kwargs)\n",
      "/home/pd378/miniconda3/envs/jupyter_env/lib/python3.6/site-packages/ipykernel_launcher.py:34: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unable to open blast table 005-B313_1-CTX\n",
      "Unable to open blast table 006-B313_1-CTX\n",
      "Unable to open blast table 007-B313_1-CTX\n",
      "Unable to open blast table 008-B313_1-CTX\n",
      "Unable to open blast table 009-B313_1-SHV\n",
      "Unable to open blast table 010-B313_1-SHV\n",
      "Unable to open blast table 011-B313_1-SHV\n",
      "Unable to open blast table 012-B313_1-SHV\n",
      "Unable to open blast table 013-B313_1-TEM\n",
      "Unable to open blast table 014-B313_1-TEM\n",
      "Unable to open blast table 015-B313_1-TEM\n",
      "Unable to open blast table 016-B313_1-TEM\n",
      "Unable to open blast table 021-B314_2-CTX\n",
      "Unable to open blast table 022-B314_2-CTX\n",
      "Unable to open blast table 023-B314_2-CTX\n",
      "Unable to open blast table 024-B314_2-CTX\n",
      "Unable to open blast table 025-B314_2-SHV\n",
      "Unable to open blast table 026-B314_2-SHV\n",
      "Unable to open blast table 027-B314_2-SHV\n",
      "Unable to open blast table 028-B314_2-SHV\n",
      "Unable to open blast table 029-B314_2-TEM\n",
      "Unable to open blast table 030-B314_2-TEM\n",
      "Unable to open blast table 031-B314_2-TEM\n",
      "Unable to open blast table 032-B314_2-TEM\n",
      "Unable to open blast table 037-B335_1-CTX\n",
      "Unable to open blast table 038-B335_1-CTX\n",
      "Unable to open blast table 039-B335_1-CTX\n",
      "Unable to open blast table 040-B335_1-CTX\n",
      "Unable to open blast table 041-B335_1-SHV\n",
      "Unable to open blast table 042-B335_1-SHV\n",
      "Unable to open blast table 043-B335_1-SHV\n",
      "Unable to open blast table 044-B335_1-SHV\n",
      "Unable to open blast table 045-B335_1-TEM\n",
      "Unable to open blast table 046-B335_1-TEM\n",
      "Unable to open blast table 047-B335_1-TEM\n",
      "Unable to open blast table 048-B335_1-TEM\n",
      "Unable to open blast table 053-B335_2-CTX\n",
      "Unable to open blast table 054-B335_2-CTX\n",
      "Unable to open blast table 055-B335_2-CTX\n",
      "Unable to open blast table 056-B335_2-CTX\n",
      "Unable to open blast table 057-B335_2-SHV\n",
      "Unable to open blast table 058-B335_2-SHV\n",
      "Unable to open blast table 059-B335_2-SHV\n",
      "Unable to open blast table 060-B335_2-SHV\n",
      "Unable to open blast table 061-B335_2-TEM\n",
      "Unable to open blast table 062-B335_2-TEM\n",
      "Unable to open blast table 063-B335_2-TEM\n",
      "Unable to open blast table 064-B335_2-TEM\n",
      "Unable to open blast table 069-B335_3-CTX\n",
      "Unable to open blast table 070-B335_3-CTX\n",
      "Unable to open blast table 071-B335_3-CTX\n",
      "Unable to open blast table 072-B335_3-CTX\n",
      "Unable to open blast table 073-B335_3-SHV\n",
      "Unable to open blast table 074-B335_3-SHV\n",
      "Unable to open blast table 075-B335_3-SHV\n",
      "Unable to open blast table 076-B335_3-SHV\n",
      "Unable to open blast table 077-B335_3-TEM\n",
      "Unable to open blast table 078-B335_3-TEM\n",
      "Unable to open blast table 079-B335_3-TEM\n",
      "Unable to open blast table 080-B335_3-TEM\n",
      "Unable to open blast table 091-R1_NEG-CTX\n",
      "Unable to open blast table 092-R1_NEG-CTX\n",
      "Unable to open blast table 093-R1_NEG-CTX\n",
      "Unable to open blast table 094-R1_NEG-CTX\n",
      "Unable to open blast table 095-R1_NEG-SHV\n",
      "Unable to open blast table 096-R1_NEG-SHV\n",
      "Unable to open blast table 097-R1_NEG-SHV\n",
      "Unable to open blast table 098-R1_NEG-SHV\n",
      "Unable to open blast table 099-R1_NEG-TEM\n",
      "Unable to open blast table 100-R1_NEG-TEM\n",
      "Unable to open blast table 101-R1_NEG-TEM\n",
      "Unable to open blast table 102-R1_NEG-TEM\n",
      "Unable to open blast table 104-R2_NEG-CTX\n",
      "Unable to open blast table 105-R2_NEG-SHV\n",
      "Unable to open blast table 106-R2_NEG-TEM\n",
      "Unable to open blast table 112-UC02_Spike-CTX\n",
      "Unable to open blast table 113-UC02_Spike-CTX\n",
      "Unable to open blast table 114-UC02_Spike-CTX\n",
      "Unable to open blast table 115-UC02_Spike-CTX\n",
      "Unable to open blast table 116-UC02_Spike-SHV\n",
      "Unable to open blast table 117-UC02_Spike-SHV\n",
      "Unable to open blast table 118-UC02_Spike-SHV\n",
      "Unable to open blast table 119-UC02_Spike-SHV\n",
      "Unable to open blast table 120-UC02_Spike-TEM\n",
      "Unable to open blast table 121-UC02_Spike-TEM\n",
      "Unable to open blast table 122-UC02_Spike-TEM\n",
      "Unable to open blast table 123-UC02_Spike-TEM\n",
      "Unable to open blast table 128-UC02_Spike-CTX\n",
      "Unable to open blast table 129-UC02_Spike-CTX\n",
      "Unable to open blast table 130-UC02_Spike-CTX\n",
      "Unable to open blast table 131-UC02_Spike-CTX\n",
      "Unable to open blast table 132-UC02_Spike-SHV\n",
      "Unable to open blast table 133-UC02_Spike-SHV\n",
      "Unable to open blast table 134-UC02_Spike-SHV\n",
      "Unable to open blast table 135-UC02_Spike-SHV\n",
      "Unable to open blast table 136-UC02_Spike-TEM\n",
      "Unable to open blast table 137-UC02_Spike-TEM\n",
      "Unable to open blast table 138-UC02_Spike-TEM\n",
      "Unable to open blast table 139-UC02_Spike-TEM\n",
      "Unable to open blast table 073-spike-H01-0p\n",
      "Unable to open blast table 074-spike-H01-0-1p\n",
      "Unable to open blast table 075-spike-H01-1p\n",
      "Unable to open blast table 076-spike-H01-10p\n",
      "Unable to open blast table 077-spike-H02-0p\n",
      "Unable to open blast table 078-spike-H02-0-1p\n",
      "Unable to open blast table 079-spike-H02-1p\n",
      "Unable to open blast table 080-spike-H02-10p\n",
      "Unable to open blast table 081-spike-H01-0p\n",
      "Unable to open blast table 082-spike-H01-0-1p\n",
      "Unable to open blast table 083-spike-H01-1p\n",
      "Unable to open blast table 084-spike-H01-10p\n",
      "Unable to open blast table 085-spike-H02-0p\n",
      "Unable to open blast table 086-spike-H02-0-1p\n",
      "Unable to open blast table 087-spike-H02-1p\n",
      "Unable to open blast table 088-spike-H02-10p\n",
      "Unable to open blast table 089-spike-H01-0p\n",
      "Unable to open blast table 090-spike-H01-0-1p\n",
      "Unable to open blast table 091-spike-H01-1p\n",
      "Unable to open blast table 092-spike-H01-10p\n",
      "Unable to open blast table 093-spike-H02-0p\n",
      "Unable to open blast table 094-spike-H02-0-1p\n",
      "Unable to open blast table 095-spike-H02-1p\n",
      "Unable to open blast table 096-spike-H02-10p\n",
      "Unable to open blast table 097-spike-C02-0p\n",
      "Unable to open blast table 098-spike-C02-0-1p\n",
      "Unable to open blast table 099-spike-C02-1p\n",
      "Unable to open blast table 100-spike-C02-10p\n",
      "Unable to open blast table 102-spike-NEG-R1\n",
      "Unable to open blast table 103-spike-H01-pSpike\n",
      "Unable to open blast table 104-spike-NEG-R2\n",
      "Unable to open blast table 105-spike-C02-0p\n",
      "Unable to open blast table 106-spike-C02-0-1p\n",
      "Unable to open blast table 107-spike-C02-1p\n",
      "Unable to open blast table 108-spike-C02-10p\n",
      "Unable to open blast table 110-spike-NEG-R1\n",
      "Unable to open blast table 111-spike-H02-pSpike\n",
      "Unable to open blast table 112-spike-NEG-R2\n",
      "Unable to open blast table 113-spike-C02-0p\n",
      "Unable to open blast table 114-spike-C02-0-1p\n",
      "Unable to open blast table 115-spike-C02-1p\n",
      "Unable to open blast table 116-spike-C02-10p\n",
      "Unable to open blast table 118-spike-NEG-R1\n",
      "Unable to open blast table 119-spike-C02-pSpike\n",
      "Unable to open blast table 142-Taxa-C02-vulg\n"
     ]
    }
   ],
   "source": [
    "with open(F_LIST) as f:\n",
    "    for line in f: \n",
    "        NAME = line.strip()\n",
    "        b_tab=BLAST_TABS+NAME+'_target.blast'\n",
    "        \n",
    "        try:\n",
    "            with open(b_tab) as fh:\n",
    "                pass\n",
    "        except IOError as e:\n",
    "            print (\"Unable to open blast table \"+NAME)\n",
    "            continue\n",
    "\n",
    "        #modify the index names for the blast table \n",
    "        blast_df = pd.read_csv(b_tab, sep='\\t', names=['p.ident','other','length','mismatch','gapopen','q.start','q.end','s.start','s.send','eval','bitscore','s.title'])\n",
    "        index_list=[]\n",
    "        for item in blast_df.index:\n",
    "            index_list.append(item.strip(';'))\n",
    "        blast_df.index=index_list\n",
    "\n",
    "        #reformat the blast information so it's in a single cell\n",
    "        blast_df['blast_summary']=blast_df['other'].astype(str)+\";\"+blast_df['length'].astype(str)+';'+blast_df['mismatch'].astype(str)+\";\"+blast_df['gapopen'].astype(str)+\";\"+blast_df['q.start'].astype(str)+\";\"+blast_df['q.end'].astype(str)+\";\"+blast_df['s.start'].astype(str)+\";\"+blast_df['s.send'].astype(str)+\";\"+blast_df['eval'].astype(str)\n",
    "        uniq_list=[]\n",
    "        size_list=[]\n",
    "        genus_list=[]\n",
    "        for item in blast_df.index:\n",
    "            uniq_list.append(item.split(\";\")[0])\n",
    "            size_list.append(item.split(\";\")[1].strip(\"size=\"))\n",
    "        blast_df['blast_uniq']=uniq_list\n",
    "        blast_df['blast_size']=size_list\n",
    "\n",
    "        #clean up the table\n",
    "        blast_clean=blast_df[['s.title','blast_summary','blast_uniq','blast_size']]\n",
    "        blast_clean.rename(columns = {'s.title':'blast_hit'}, inplace = True)\n",
    "        blast_clean[\"file\"]=NAME\n",
    "        \n",
    "        #merge the taxa information with the blast information\n",
    "        final_df=pd.merge(blast_clean, clean_df,left_on=['file', 'blast_uniq'], right_on=['file','in_uniq'], how='left' )\n",
    "        \n",
    "        #format and condens the final table for export\n",
    "        final_df=final_df.fillna(\"0\")\n",
    "        final_df['in_size']=final_df['in_size'].astype(int)\n",
    "        final_df['sum_size']=final_df.groupby(by=['Phylum', 'Class', 'Order','Family','Genus',\"blast_hit\"])[\"in_size\"].transform(sum)\n",
    "        reduce_df=final_df.drop_duplicates(subset=['Phylum', 'Class', 'Order','Family', 'Genus',\"blast_hit\"]).sort_values(by=['sum_size'],ascending=False)\n",
    "        reduce_df=reduce_df.drop(['blast_uniq','in_uniq','blast_size','in_size'], axis=1)\n",
    "        reduce_df.to_csv(OUT_TAB+NAME+\"_combined_tab.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#004-B313_1-Bact_target.blast\n",
    "blast_df = pd.read_csv(BLAST_TABS+'004-B313_1-Bact_target.blast', sep='\\t', names=['p.ident','other','length','mismatch','gapopen','q.start','q.end','s.start','s.send','eval','bitscore','s.title'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>p.ident</th>\n",
       "      <th>other</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>q.start</th>\n",
       "      <th>q.end</th>\n",
       "      <th>s.start</th>\n",
       "      <th>s.send</th>\n",
       "      <th>eval</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>s.title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Uniq1;size=890;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2;size=865;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq3;size=331;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq4;size=295;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq5;size=214;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq6;size=211;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq7;size=141;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq8;size=135;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq9;size=127;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq10;size=97;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq11;size=74;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq12;size=65;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq13;size=64;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq14;size=59;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq15;size=59;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq16;size=57;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq17;size=55;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq18;size=54;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq19;size=49;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq20;size=48;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq21;size=47;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq22;size=45;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq23;size=45;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq24;size=41;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq25;size=40;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq26;size=39;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq27;size=39;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq28;size=38;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq29;size=37;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq30;size=34;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2963;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>99.057</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>3.360000e-45</td>\n",
       "      <td>191.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2964;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>99.048</td>\n",
       "      <td>105</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>105</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816464</td>\n",
       "      <td>1.210000e-44</td>\n",
       "      <td>189.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2965;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>105</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>105</td>\n",
       "      <td>3816567</td>\n",
       "      <td>3816463</td>\n",
       "      <td>2.560000e-46</td>\n",
       "      <td>195.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2966;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2967;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2968;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2969;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2970;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2971;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>98.113</td>\n",
       "      <td>106</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>1.560000e-43</td>\n",
       "      <td>185.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2972;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>96.939</td>\n",
       "      <td>98</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>106</td>\n",
       "      <td>3816560</td>\n",
       "      <td>3816463</td>\n",
       "      <td>2.040000e-37</td>\n",
       "      <td>165.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2973;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>99.057</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>3.360000e-45</td>\n",
       "      <td>191.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2974;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>98.113</td>\n",
       "      <td>106</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>1.560000e-43</td>\n",
       "      <td>185.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2975;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>98.095</td>\n",
       "      <td>105</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>105</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816464</td>\n",
       "      <td>5.630000e-43</td>\n",
       "      <td>183.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2976;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2977;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2978;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2979;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>99.057</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>3.360000e-45</td>\n",
       "      <td>191.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2980;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2981;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2982;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>99.057</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>3.360000e-45</td>\n",
       "      <td>191.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2983;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2984;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2985;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>99.057</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>3.360000e-45</td>\n",
       "      <td>191.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2986;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2987;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>94.340</td>\n",
       "      <td>106</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.330000e-37</td>\n",
       "      <td>163.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2988;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2989;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2990;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>105</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>105</td>\n",
       "      <td>505684</td>\n",
       "      <td>505788</td>\n",
       "      <td>2.560000e-46</td>\n",
       "      <td>195.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2991;size=1;</th>\n",
       "      <td>gi|1042799677|gb|CP012706.1|</td>\n",
       "      <td>100.000</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>3816568</td>\n",
       "      <td>3816463</td>\n",
       "      <td>7.230000e-47</td>\n",
       "      <td>196.0</td>\n",
       "      <td>Bacteroides fragilis strain S14, complete genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Uniq2992;size=1;</th>\n",
       "      <td>gi|938475369|gb|CP012937.1|</td>\n",
       "      <td>94.340</td>\n",
       "      <td>106</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>106</td>\n",
       "      <td>505683</td>\n",
       "      <td>505788</td>\n",
       "      <td>7.330000e-37</td>\n",
       "      <td>163.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron strain 7330, comp...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2991 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       p.ident    other  length  mismatch  \\\n",
       "Uniq1;size=890;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2;size=865;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq3;size=331;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq4;size=295;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq5;size=214;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq6;size=211;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq7;size=141;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq8;size=135;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq9;size=127;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq10;size=97;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq11;size=74;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq12;size=65;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq13;size=64;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq14;size=59;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq15;size=59;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq16;size=57;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq17;size=55;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq18;size=54;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq19;size=49;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq20;size=48;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq21;size=47;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq22;size=45;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq23;size=45;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq24;size=41;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq25;size=40;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq26;size=39;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq27;size=39;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq28;size=38;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq29;size=37;   gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq30;size=34;    gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "...                                        ...      ...     ...       ...   \n",
       "Uniq2963;size=1;   gi|938475369|gb|CP012937.1|   99.057     106         1   \n",
       "Uniq2964;size=1;  gi|1042799677|gb|CP012706.1|   99.048     105         1   \n",
       "Uniq2965;size=1;  gi|1042799677|gb|CP012706.1|  100.000     105         0   \n",
       "Uniq2966;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2967;size=1;   gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq2968;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2969;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2970;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2971;size=1;   gi|938475369|gb|CP012937.1|   98.113     106         2   \n",
       "Uniq2972;size=1;  gi|1042799677|gb|CP012706.1|   96.939      98         3   \n",
       "Uniq2973;size=1;  gi|1042799677|gb|CP012706.1|   99.057     106         1   \n",
       "Uniq2974;size=1;   gi|938475369|gb|CP012937.1|   98.113     106         2   \n",
       "Uniq2975;size=1;  gi|1042799677|gb|CP012706.1|   98.095     105         2   \n",
       "Uniq2976;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2977;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2978;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2979;size=1;  gi|1042799677|gb|CP012706.1|   99.057     106         1   \n",
       "Uniq2980;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2981;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2982;size=1;   gi|938475369|gb|CP012937.1|   99.057     106         1   \n",
       "Uniq2983;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2984;size=1;   gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq2985;size=1;   gi|938475369|gb|CP012937.1|   99.057     106         1   \n",
       "Uniq2986;size=1;   gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq2987;size=1;  gi|1042799677|gb|CP012706.1|   94.340     106         6   \n",
       "Uniq2988;size=1;   gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq2989;size=1;   gi|938475369|gb|CP012937.1|  100.000     106         0   \n",
       "Uniq2990;size=1;   gi|938475369|gb|CP012937.1|  100.000     105         0   \n",
       "Uniq2991;size=1;  gi|1042799677|gb|CP012706.1|  100.000     106         0   \n",
       "Uniq2992;size=1;   gi|938475369|gb|CP012937.1|   94.340     106         6   \n",
       "\n",
       "                  gapopen  q.start  q.end  s.start   s.send          eval  \\\n",
       "Uniq1;size=890;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2;size=865;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq3;size=331;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq4;size=295;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq5;size=214;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq6;size=211;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq7;size=141;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq8;size=135;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq9;size=127;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq10;size=97;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq11;size=74;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq12;size=65;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq13;size=64;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq14;size=59;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq15;size=59;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq16;size=57;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq17;size=55;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq18;size=54;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq19;size=49;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq20;size=48;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq21;size=47;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq22;size=45;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq23;size=45;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq24;size=41;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq25;size=40;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq26;size=39;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq27;size=39;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq28;size=38;         0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq29;size=37;         0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq30;size=34;         0        1    106   505683   505788  7.230000e-47   \n",
       "...                   ...      ...    ...      ...      ...           ...   \n",
       "Uniq2963;size=1;        0        1    106   505683   505788  3.360000e-45   \n",
       "Uniq2964;size=1;        0        1    105  3816568  3816464  1.210000e-44   \n",
       "Uniq2965;size=1;        0        1    105  3816567  3816463  2.560000e-46   \n",
       "Uniq2966;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2967;size=1;        0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq2968;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2969;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2970;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2971;size=1;        0        1    106   505683   505788  1.560000e-43   \n",
       "Uniq2972;size=1;        0        9    106  3816560  3816463  2.040000e-37   \n",
       "Uniq2973;size=1;        0        1    106  3816568  3816463  3.360000e-45   \n",
       "Uniq2974;size=1;        0        1    106   505683   505788  1.560000e-43   \n",
       "Uniq2975;size=1;        0        1    105  3816568  3816464  5.630000e-43   \n",
       "Uniq2976;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2977;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2978;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2979;size=1;        0        1    106  3816568  3816463  3.360000e-45   \n",
       "Uniq2980;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2981;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2982;size=1;        0        1    106   505683   505788  3.360000e-45   \n",
       "Uniq2983;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2984;size=1;        0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq2985;size=1;        0        1    106   505683   505788  3.360000e-45   \n",
       "Uniq2986;size=1;        0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq2987;size=1;        0        1    106  3816568  3816463  7.330000e-37   \n",
       "Uniq2988;size=1;        0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq2989;size=1;        0        1    106   505683   505788  7.230000e-47   \n",
       "Uniq2990;size=1;        0        1    105   505684   505788  2.560000e-46   \n",
       "Uniq2991;size=1;        0        1    106  3816568  3816463  7.230000e-47   \n",
       "Uniq2992;size=1;        0        1    106   505683   505788  7.330000e-37   \n",
       "\n",
       "                  bitscore                                            s.title  \n",
       "Uniq1;size=890;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2;size=865;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq3;size=331;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq4;size=295;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq5;size=214;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq6;size=211;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq7;size=141;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq8;size=135;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq9;size=127;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq10;size=97;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq11;size=74;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq12;size=65;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq13;size=64;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq14;size=59;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq15;size=59;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq16;size=57;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq17;size=55;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq18;size=54;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq19;size=49;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq20;size=48;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq21;size=47;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq22;size=45;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq23;size=45;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq24;size=41;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq25;size=40;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq26;size=39;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq27;size=39;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq28;size=38;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq29;size=37;      196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq30;size=34;      196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "...                    ...                                                ...  \n",
       "Uniq2963;size=1;     191.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2964;size=1;     189.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2965;size=1;     195.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2966;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2967;size=1;     196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2968;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2969;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2970;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2971;size=1;     185.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2972;size=1;     165.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2973;size=1;     191.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2974;size=1;     185.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2975;size=1;     183.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2976;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2977;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2978;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2979;size=1;     191.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2980;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2981;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2982;size=1;     191.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2983;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2984;size=1;     196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2985;size=1;     191.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2986;size=1;     196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2987;size=1;     163.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2988;size=1;     196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2989;size=1;     196.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2990;size=1;     195.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "Uniq2991;size=1;     196.0   Bacteroides fragilis strain S14, complete genome  \n",
       "Uniq2992;size=1;     163.0  Bacteroides thetaiotaomicron strain 7330, comp...  \n",
       "\n",
       "[2991 rows x 12 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blast_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'146-Taxa-NR2-Prev10'"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NAME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
